### ISSUE TITLE
failed to flush the buffer(connect_write timeout reached) 

### ISSUE THREAD

**Issue by linbingdouzhe** [uken/fluent-plugin-elasticsearch#802]
(check apply)
- [ ] read [the contribution guideline](https://github.com/uken/fluent-plugin-elasticsearch/blob/master/CONTRIBUTING.md)

#### Problem

...

#### Steps to replicate

Either clone and modify https://gist.github.com/pitr/9a518e840db58f435911

**OR**

Provide example config and message

#### Expected Behavior or What you need to ask
    <filter kube.var.log.containers.coredns**>
      @type parser
      @log_level info
      format json
      key_name log
      reserve_data true
      <parse>
        @type json
        time_key time
        #utc true
        time_format %Y-%m-%dT%H:%M:%S.%9NZ
        timezone "+08:00"
      </parse>
    </filter>
    <match kube.var.log.containers.coredns**>
      @id coredns
      @type elasticsearch
      include_tag_key true
      # Replace with the host/port to your Elasticsearch cluster.
      host "opendistro.wisersops.internal"
      port "9200"
      scheme "https"
      ssl_version "TLSv1_2"
      ssl on
      ssl_verify false
      user admin
      password z1zNWKSXtUQKwOMr
      logstash_format true
      logstash_prefix ihb-aws-coredns
      <buffer>
        @type file
        path /var/log/fluentd-buffers/coredns/kubernetes.system.buffer
        flush_mode interval
        retry_type exponential_backoff
        flush_thread_count 2
        flush_interval 5s
        retry_forever
        retry_max_interval 30
        reconnect_on_error true
        reload_on_failure true
        reload_connections false
        overflow_action block
      </buffer>
    </match>
    <match kube.var.log.containers.coredns**>
      @id coredns
      @type elasticsearch
      include_tag_key true
      # Replace with the host/port to your Elasticsearch cluster.
      host "es.internal"
      port "9200"
      scheme "https"
      ssl_version "TLSv1_2"
      ssl on
      ssl_verify false
      user admin
      password ***
      logstash_format true
      logstash_prefix aws-coredns
      <buffer>
        @type file
        path /var/log/fluentd-buffers/coredns/kubernetes.system.buffer
        flush_mode interval
        retry_type exponential_backoff
        flush_thread_count 2
        flush_interval 5s
        retry_forever
        retry_max_interval 30
        reconnect_on_error true
        reload_on_failure true
        reload_connections false
        overflow_action block
      </buffer>
    </match>
...

#### Using Fluentd and ES plugin versions

* OS version
rancheros v1.5.4
* Bare Metal or within Docker or Kubernetes or others? 
Kubernetes
* Fluentd v0.12 or v0.14/v1.0
  * paste result of ``fluentd --version`` or ``td-agent --version``
fluentd 1.3.3
* ES plugin 3.x.y/2.x.y or 1.x.y
  * paste boot log of fluentd or td-agent
```
retry_type exponential_backoff
flush_thread_count 2
flush_interval 5s
retry_forever
retry_max_interval 30
reconnect_on_error true
reload_on_failure true
reload_connections false
overflow_action block
</buffer>
</match> is not used.
2020-09-08 03:04:54 +0000 [warn]: parameter 'reconnect_on_error' in <buffer>
@type "file"
path "/var/log/fluentd-buffers/dma-dolphin-solr-service/kubernetes.system.buffer"
flush_mode interval
retry_type exponential_backoff
flush_thread_count 2
flush_interval 5s
retry_forever
retry_max_interval 30
reconnect_on_error true
reload_on_failure true
reload_connections false
overflow_action block
</buffer> is not used.
2020-09-08 03:04:54 +0000 [warn]: parameter 'reload_on_failure' in <buffer>
@type "file"
path "/var/log/fluentd-buffers/dma-dolphin-solr-service/kubernetes.system.buffer"
flush_mode interval
retry_type exponential_backoff
flush_thread_count 2
flush_interval 5s
retry_forever
retry_max_interval 30
reconnect_on_error true
reload_on_failure true
reload_connections false
overflow_action block
</buffer> is not used.
2020-09-08 03:04:54 +0000 [warn]: parameter 'reload_connections' in <buffer>
@type "file"
path "/var/log/fluentd-buffers/dma-dolphin-solr-service/kubernetes.system.buffer"
flush_mode interval
retry_type exponential_backoff
flush_thread_count 2
flush_interval 5s
retry_forever
retry_max_interval 30
reconnect_on_error true
reload_on_failure true
reload_connections false
overflow_action block
</buffer> is not used.
2020-09-08 03:04:54 +0000 [warn]: parameter 'ssl' in <match kube.var.log.containers.dma-config-service**_production_**>
@id dma-config-service
@type elasticsearch
include_tag_key true
host "opendistro.wisersops.internal"
port 9200
scheme https
user "admin"
password xxxxxx
ssl_version TLSv1_2
ssl on
ssl_verify false
logstash_format true
logstash_prefix "dma-config-service-prod"
<buffer>
@type "file"
path "/var/log/fluentd-buffers/dma-config-service/kubernetes.system.buffer"
flush_mode interval
retry_type exponential_backoff
flush_thread_count 2
flush_interval 5s
retry_forever
retry_max_interval 30
reconnect_on_error true
reload_on_failure true
reload_connections false
overflow_action block
</buffer>
</match> is not used.
2020-09-08 03:04:54 +0000 [warn]: parameter 'reconnect_on_error' in <buffer>
@type "file"
path "/var/log/fluentd-buffers/dma-config-service/kubernetes.system.buffer"
flush_mode interval
retry_type exponential_backoff
flush_thread_count 2
flush_interval 5s
retry_forever
retry_max_interval 30
reconnect_on_error true
reload_on_failure true
reload_connections false
overflow_action block
</buffer> is not used.
2020-09-08 03:04:54 +0000 [warn]: parameter 'reload_on_failure' in <buffer>
@type "file"
path "/var/log/fluentd-buffers/dma-config-service/kubernetes.system.buffer"
flush_mode interval
retry_type exponential_backoff
flush_thread_count 2
flush_interval 5s
retry_forever
retry_max_interval 30
reconnect_on_error true
reload_on_failure true
reload_connections false
overflow_action block
</buffer> is not used.
2020-09-08 03:04:54 +0000 [warn]: parameter 'reload_connections' in <buffer>
@type "file"
path "/var/log/fluentd-buffers/dma-config-service/kubernetes.system.buffer"
flush_mode interval
retry_type exponential_backoff
flush_thread_count 2
flush_interval 5s
retry_forever
retry_max_interval 30
reconnect_on_error true
reload_on_failure true
reload_connections false
overflow_action block
</buffer> is not used.
2020-09-08 03:04:54 +0000 [warn]: parameter 'ssl' in <match kube.var.log.containers.dma-user-behavior-service**_production_**>
@id dma-user-behavior-service
@type elasticsearch
include_tag_key true
host "opendistro.wisersops.internal"
port 9200
scheme https
user "admin"
password xxxxxx
ssl_version TLSv1_2
ssl on
ssl_verify false
logstash_format true
logstash_prefix "dma-user-behavior-service-prod"
<buffer>
@type "file"
path "/var/log/fluentd-buffers/dma-user-behavior-service/kubernetes.system.buffer"
flush_mode interval
retry_type exponential_backoff
flush_thread_count 2
flush_interval 5s
retry_forever
retry_max_interval 30
reconnect_on_error true
reload_on_failure true
reload_connections false
overflow_action block
</buffer>
</match> is not used.
2020-09-08 03:04:54 +0000 [warn]: parameter 'reconnect_on_error' in <buffer>
@type "file"
path "/var/log/fluentd-buffers/dma-user-behavior-service/kubernetes.system.buffer"
flush_mode interval
retry_type exponential_backoff
flush_thread_count 2
flush_interval 5s
retry_forever
retry_max_interval 30
reconnect_on_error true
reload_on_failure true
reload_connections false
overflow_action block
</buffer> is not used.
2020-09-08 03:04:54 +0000 [warn]: parameter 'reload_on_failure' in <buffer>
@type "file"
path "/var/log/fluentd-buffers/dma-user-behavior-service/kubernetes.system.buffer"
flush_mode interval
retry_type exponential_backoff
flush_thread_count 2
flush_interval 5s
retry_forever
retry_max_interval 30
reconnect_on_error true
reload_on_failure true
reload_connections false
overflow_action block
</buffer> is not used.
2020-09-08 03:04:54 +0000 [warn]: parameter 'reload_connections' in <buffer>
@type "file"
path "/var/log/fluentd-buffers/dma-user-behavior-service/kubernetes.system.buffer"
flush_mode interval
retry_type exponential_backoff
flush_thread_count 2
flush_interval 5s
retry_forever
retry_max_interval 30
reconnect_on_error true
reload_on_failure true
reload_connections false
overflow_action block
</buffer> is not used.
2020-09-08 03:04:54 +0000 [warn]: parameter 'ssl' in <match kube.var.log.containers.dma-user-service**_production_**>
@id dma-user-service
@type elasticsearch
include_tag_key true
host "opendistro.wisersops.internal"
port 9200
scheme https
user "admin"
password xxxxxx
ssl_version TLSv1_2
ssl on
ssl_verify false
logstash_format true
logstash_prefix "dma-user-service-prod"
<buffer>
@type "file"
path "/var/log/fluentd-buffers/dma-user-service/kubernetes.system.buffer"
flush_mode interval
retry_type exponential_backoff
flush_thread_count 2
flush_interval 5s
retry_forever
retry_max_interval 30
reconnect_on_error true
reload_on_failure true
reload_connections false
overflow_action block
</buffer>
</match> is not used.
2020-09-08 03:04:54 +0000 [warn]: parameter 'reconnect_on_error' in <buffer>
@type "file"
path "/var/log/fluentd-buffers/dma-user-service/kubernetes.system.buffer"
flush_mode interval
retry_type exponential_backoff
flush_thread_count 2
flush_interval 5s
retry_forever
retry_max_interval 30
reconnect_on_error true
reload_on_failure true
reload_connections false
overflow_action block
</buffer> is not used.
2020-09-08 03:04:54 +0000 [warn]: parameter 'reload_on_failure' in <buffer>
@type "file"
path "/var/log/fluentd-buffers/dma-user-service/kubernetes.system.buffer"
flush_mode interval
retry_type exponential_backoff
flush_thread_count 2
flush_interval 5s
retry_forever
retry_max_interval 30
reconnect_on_error true
reload_on_failure true
reload_connections false
overflow_action block
</buffer> is not used.
2020-09-08 03:04:54 +0000 [warn]: parameter 'reload_connections' in <buffer>
@type "file"
path "/var/log/fluentd-buffers/dma-user-service/kubernetes.system.buffer"
flush_mode interval
retry_type exponential_backoff
flush_thread_count 2
flush_interval 5s
retry_forever
retry_max_interval 30
reconnect_on_error true
reload_on_failure true
reload_connections false
overflow_action block
</buffer> is not used.
2020-09-08 03:04:54 +0000 [warn]: parameter 'ssl' in <match kube.var.log.containers.dma-kg-service**_production_**>
@id dma-kg-service
@type elasticsearch
include_tag_key true
host "opendistro.wisersops.internal"
port 9200
scheme https
user "admin"
password xxxxxx
ssl_version TLSv1_2
ssl on
ssl_verify false
logstash_format true
logstash_prefix "dma-kg-service-prod"
<buffer>
@type "file"
path "/var/log/fluentd-buffers/dma-kg-service/kubernetes.system.buffer"
flush_mode interval
retry_type exponential_backoff
flush_thread_count 2
flush_interval 5s
retry_forever
retry_max_interval 30
reconnect_on_error true
reload_on_failure true
reload_connections false
overflow_action block
</buffer>
</match> is not used.
2020-09-08 03:04:54 +0000 [warn]: parameter 'reconnect_on_error' in <buffer>
@type "file"
path "/var/log/fluentd-buffers/dma-kg-service/kubernetes.system.buffer"
flush_mode interval
retry_type exponential_backoff
flush_thread_count 2
flush_interval 5s
retry_forever
retry_max_interval 30
reconnect_on_error true
reload_on_failure true
reload_connections false
overflow_action block
</buffer> is not used.
2020-09-08 03:04:54 +0000 [warn]: parameter 'reload_on_failure' in <buffer>
@type "file"
path "/var/log/fluentd-buffers/dma-kg-service/kubernetes.system.buffer"
flush_mode interval
retry_type exponential_backoff
flush_thread_count 2
flush_interval 5s
retry_forever
retry_max_interval 30
reconnect_on_error true
reload_on_failure true
reload_connections false
overflow_action block
</buffer> is not used.
2020-09-08 03:04:54 +0000 [warn]: parameter 'reload_connections' in <buffer>
@type "file"
path "/var/log/fluentd-buffers/dma-kg-service/kubernetes.system.buffer"
flush_mode interval
retry_type exponential_backoff
flush_thread_count 2
flush_interval 5s
retry_forever
retry_max_interval 30
reconnect_on_error true
reload_on_failure true
reload_connections false
overflow_action block
</buffer> is not used.
2020-09-08 03:04:54 +0000 [warn]: parameter 'ssl' in <match kube.var.log.containers.dma-api-gateway**_production_**>
@id dma-api-gateway
@type elasticsearch
include_tag_key true
host "opendistro.wisersops.internal"
port 9200
scheme https
user "admin"
password xxxxxx
ssl_version TLSv1_2
ssl on
ssl_verify false
logstash_format true
logstash_prefix "dma-api-gateway-prod"
<buffer>
@type "file"
path "/var/log/fluentd-buffers/dma-api-gateway/kubernetes.system.buffer"
flush_mode interval
retry_type exponential_backoff
flush_thread_count 2
flush_interval 5s
retry_forever
retry_max_interval 30
reconnect_on_error true
reload_on_failure true
reload_connections false
overflow_action block
</buffer>
</match> is not used.
2020-09-08 03:04:54 +0000 [warn]: parameter 'reconnect_on_error' in <buffer>
@type "file"
path "/var/log/fluentd-buffers/dma-api-gateway/kubernetes.system.buffer"
flush_mode interval
retry_type exponential_backoff
flush_thread_count 2
flush_interval 5s
retry_forever
retry_max_interval 30
reconnect_on_error true
reload_on_failure true
reload_connections false
overflow_action block
</buffer> is not used.
2020-09-08 03:04:54 +0000 [warn]: parameter 'reload_on_failure' in <buffer>
@type "file"
path "/var/log/fluentd-buffers/dma-api-gateway/kubernetes.system.buffer"
flush_mode interval
retry_type exponential_backoff
flush_thread_count 2
flush_interval 5s
retry_forever
retry_max_interval 30
reconnect_on_error true
reload_on_failure true
reload_connections false
overflow_action block
</buffer> is not used.
2020-09-08 03:04:54 +0000 [warn]: parameter 'reload_connections' in <buffer>
@type "file"
path "/var/log/fluentd-buffers/dma-api-gateway/kubernetes.system.buffer"
flush_mode interval
retry_type exponential_backoff
flush_thread_count 2
flush_interval 5s
retry_forever
retry_max_interval 30
reconnect_on_error true
reload_on_failure true
reload_connections false
overflow_action block
</buffer> is not used.
2020-09-08 03:04:54 +0000 [warn]: parameter 'reconnect_on_error' in <buffer>
@type "file"
path "/var/log/fluentd-buffers/apm-dma-dolphin-plus-service/kubernetes.system.buffer"
flush_mode interval
retry_type exponential_backoff
flush_thread_count 2
flush_interval 5s
retry_forever
retry_max_interval 30
reconnect_on_error true
reload_on_failure true
reload_connections false
overflow_action block
</buffer> is not used.
2020-09-08 03:04:54 +0000 [warn]: parameter 'reload_on_failure' in <buffer>
@type "file"
path "/var/log/fluentd-buffers/apm-dma-dolphin-plus-service/kubernetes.system.buffer"
flush_mode interval
retry_type exponential_backoff
flush_thread_count 2
flush_interval 5s
retry_forever
retry_max_interval 30
reconnect_on_error true
reload_on_failure true
reload_connections false
overflow_action block
</buffer> is not used.
2020-09-08 03:04:54 +0000 [warn]: parameter 'reload_connections' in <buffer>
@type "file"
path "/var/log/fluentd-buffers/apm-dma-dolphin-plus-service/kubernetes.system.buffer"
flush_mode interval
retry_type exponential_backoff
flush_thread_count 2
flush_interval 5s
retry_forever
retry_max_interval 30
reconnect_on_error true
reload_on_failure true
reload_connections false
overflow_action block
</buffer> is not used.
2020-09-08 03:04:54 +0000 [info]: #0 starting fluentd worker pid=10 ppid=1 worker=0
2020-09-08 03:04:54 +0000 [debug]: #0 [apm-dma-dolphin-plus] buffer started instance=70272220784340 stage_size=0 queue_size=0
2020-09-08 03:04:54 +0000 [debug]: #0 [dma-api-gateway] buffer started instance=70272224795920 stage_size=0 queue_size=0
2020-09-08 03:04:54 +0000 [debug]: #0 [dma-kg-service] buffer started instance=70272224552200 stage_size=0 queue_size=0
2020-09-08 03:04:54 +0000 [debug]: #0 [dma-user-service] buffer started instance=70272224332320 stage_size=0 queue_size=0
2020-09-08 03:04:54 +0000 [debug]: #0 [dma-user-behavior-service] buffer started instance=70272224124840 stage_size=0 queue_size=0
2020-09-08 03:04:54 +0000 [debug]: #0 [dma-config-service] buffer started instance=70272223898980 stage_size=0 queue_size=0
2020-09-08 03:04:54 +0000 [debug]: #0 [dma-dolphin-solr-service] buffer started instance=70272223709340 stage_size=0 queue_size=0
2020-09-08 03:04:54 +0000 [debug]: #0 [dma-dolphin-plus-service] buffer started instance=70272223523560 stage_size=0 queue_size=0
2020-09-08 03:04:54 +0000 [debug]: #0 [dma-coral-service] buffer started instance=70272270561900 stage_size=0 queue_size=0
2020-09-08 03:04:54 +0000 [debug]: #0 [dma-search-service] buffer started instance=70272270863540 stage_size=0 queue_size=0
2020-09-08 03:04:54 +0000 [debug]: #0 [dma-orca-service] buffer started instance=70272243798640 stage_size=0 queue_size=0
2020-09-08 03:04:54 +0000 [debug]: #0 listening monitoring http server on http://0.0.0.0:24220/api/plugins for worker0
2020-09-08 03:04:54 +0000 [debug]: #0 tag parameter is specified. Emit plugins info to 'fluentd.monitor.metrics'
2020-09-08 03:04:54 +0000 [debug]: #0 [dma-user-service] flush_thread actually running
2020-09-08 03:04:54 +0000 [debug]: #0 [dma-api-gateway] flush_thread actually running
2020-09-08 03:04:54 +0000 [debug]: #0 [dma-user-behavior-service] flush_thread actually running
2020-09-08 03:04:54 +0000 [debug]: #0 [dma-user-behavior-service] enqueue_thread actually running
2020-09-08 03:04:54 +0000 [debug]: #0 [dma-config-service] flush_thread actually running
2020-09-08 03:04:54 +0000 [debug]: #0 [dma-config-service] enqueue_thread actually running
2020-09-08 03:04:54 +0000 [debug]: #0 [dma-dolphin-solr-service] flush_thread actually running
2020-09-08 03:04:54 +0000 [debug]: #0 [dma-dolphin-solr-service] enqueue_thread actually running
2020-09-08 03:04:54 +0000 [debug]: #0 [dma-dolphin-plus-service] flush_thread actually running
2020-09-08 03:04:54 +0000 [debug]: #0 [dma-dolphin-plus-service] flush_thread actually running
2020-09-08 03:04:54 +0000 [debug]: #0 [dma-kg-service] flush_thread actually running
2020-09-08 03:04:54 +0000 [debug]: #0 [dma-coral-service] enqueue_thread actually running
2020-09-08 03:04:54 +0000 [debug]: #0 [dma-search-service] flush_thread actually running
2020-09-08 03:04:54 +0000 [debug]: #0 [dma-search-service] flush_thread actually running
2020-09-08 03:04:54 +0000 [debug]: #0 [dma-kg-service] enqueue_thread actually running
2020-09-08 03:04:54 +0000 [debug]: #0 [dma-orca-service] flush_thread actually running
2020-09-08 03:04:54 +0000 [debug]: #0 [apm-dma-dolphin-plus] flush_thread actually running
2020-09-08 03:04:54 +0000 [debug]: #0 [dma-user-service] flush_thread actually running
2020-09-08 03:04:54 +0000 [debug]: #0 [dma-user-service] enqueue_thread actually running
2020-09-08 03:04:54 +0000 [debug]: #0 [dma-user-behavior-service] flush_thread actually running
2020-09-08 03:04:54 +0000 [debug]: #0 [dma-config-service] flush_thread actually running
2020-09-08 03:04:54 +0000 [debug]: #0 [dma-dolphin-solr-service] flush_thread actually running
2020-09-08 03:04:54 +0000 [debug]: #0 [apm-dma-dolphin-plus] enqueue_thread actually running
2020-09-08 03:04:54 +0000 [debug]: #0 [dma-coral-service] flush_thread actually running
2020-09-08 03:04:54 +0000 [debug]: #0 [dma-kg-service] flush_thread actually running
2020-09-08 03:04:54 +0000 [debug]: #0 [dma-orca-service] flush_thread actually running
2020-09-08 03:04:54 +0000 [debug]: #0 [dma-orca-service] enqueue_thread actually running
2020-09-08 03:04:54 +0000 [debug]: #0 listening http bind="0.0.0.0" port=9880
2020-09-08 03:04:54 +0000 [debug]: #0 [dma-api-gateway] flush_thread actually running
2020-09-08 03:04:54 +0000 [debug]: #0 [dma-api-gateway] enqueue_thread actually running
2020-09-08 03:04:54 +0000 [debug]: #0 [dma-coral-service] flush_thread actually running
2020-09-08 03:04:54 +0000 [debug]: #0 [dma-dolphin-plus-service] enqueue_thread actually running
2020-09-08 03:04:54 +0000 [debug]: #0 [dma-search-service] enqueue_thread actually running
2020-09-08 03:04:54 +0000 [info]: #0 listening port port=24224 bind="0.0.0.0"
2020-09-08 03:04:54 +0000 [info]: #0 fluentd worker is now running worker=0
2020-09-08 03:04:54 +0000 [debug]: #0 [apm-dma-dolphin-plus] flush_thread actually running
2020-09-08 03:05:10 +0000 [debug]: #0 [dma-user-service] Created new chunk chunk_id="5aec49a61f7db6b6e0bde3e87e5c1244" metadata=#<struct Fluent::Plugin::Buffer::Metadata timekey=nil, tag="kube.var.log.containers.dma-user-service-67787785b5-clz29_production_dma-user-service-ceed805c0420f289ea7b1d0b1724a0a94834f635141919627d0a977e394c37ff.log", variables=nil>
2020-09-08 03:05:15 +0000 [warn]: #0 no patterns matched tag="kube.var.log.containers.dma-dolphin-plus-service-54776cbb57-gpj2w_be05_dma-dolphin-plus-service-9516e173a82e914f5b8876777ff9fe84dac66142590d86352f23407e9ebbde30.log"
2020-09-08 03:05:16 +0000 [warn]: #0 [dma-user-service] Detected ES 7.x or above: `_doc` will be used as the document `_type`.
2020-09-08 03:05:19 +0000 [debug]: #0 [dma-user-service] Created new chunk chunk_id="5aec49aeb47ce76109ce810cd2fd0f92" metadata=#<struct Fluent::Plugin::Buffer::Metadata timekey=nil, tag="kube.var.log.containers.dma-user-service-67787785b5-clz29_production_dma-user-service-ceed805c0420f289ea7b1d0b1724a0a94834f635141919627d0a977e394c37ff.log", variables=nil>
2020-09-08 03:05:24 +0000 [warn]: #0 [dma-user-service] Detected ES 7.x or above: `_doc` will be used as the documen
```
  * paste result of ``fluent-gem list``, ``td-agent-gem list`` or your Gemfile.lock
elasticsearch (6.1.0)

* ES version (optional)
7.3.0
* ES template(s) (optional)

**Comment by linbingdouzhe** [comment]
from #746  i get that i should use real es ip instead of es host (service name) , but i configured https for es , can not use real ip . so , any way can i aviod this ? @cosmo0920

**Comment by cosmo0920** [comment]
Fill issue template completely and write environment information (OS version, Bare Metal/Docker/Kubernetes, paste boot log or warning log).

**Comment by linbingdouzhe** [comment]
sorry for that, already updated my issue template , but this error appear to production env , i can not reproduce it , seems it will present to when fluentd running long time(maybe couple of weeks , maybe few days)

**Comment by linbingdouzhe** [comment]
any update ? is this a bug ?

**Comment by cosmo0920** [comment]
It seems that exhausted Elasticsearch cluster causes this issue.
ES plugin just sending events into there.
ES plugin and Fluentd core do not have flow control feature.
So you should pay attention to ES cluster health.

**Comment by linbingdouzhe** [comment]
i found this root cause 
because in my case , i added this  parameter
`overflow_action block`
I think when log out of limit , this set will block my log in ,
so , I deleted this parameter, now everything ok !

**Comment by linbingdouzhe** [comment]
problem occurs again ..  seem not that parameter "overflow_action" reason

**Comment by cosmo0920** [comment]
```aconf
    reconnect_on_error true
    reload_on_failure true
    reload_connections false
```

should be put under match directly instead of put in `<buffer>`.

**Comment by errogaht** [comment]
the same for me. can't solve the issue

**Comment by placydo** [comment]
Hey!
Just to let You know we had exactly same issue. None of parameter adjustment did help. After closer investigetion I did discover out that in our case (we do use logging-operator on kubernetes) the root cause of error was insufficient memory allocation. By default it had only 400mb of RAM assigned and it was riding the limit all the time. Adjusting the memory (in our case 2GB for now) resolved that issue straight away.

TLDR: assign more memory for fluentd pod

**Comment by hunghip95** [comment]
> 

I'm having fluentd running in docker, I dont apply any resource limit (it always have much free memory), the timeout error still occurs.

**Comment by loic-brtd** [comment]
> reconnect_on_error true
>     reload_on_failure true
>     reload_connections false
> should be put under match directly instead of put in `<buffer>`.

Thanks!

As strange as this config seems (especially `reload_connections false`), it solved my problem ("failed to flush the buffer(connect_write timeout reached)".

I also saw this configuration here:  https://github.com/fluent/fluent-plugin-opensearch/issues/157#issuecomment-2896986956. I don't fully understand the reasoning behind it, but this could be related.
